platform/mbed_assert.h

```diff
@@ -49,6 +49,65 @@ do {                                                     \
 } while (0)
 #endif
 
+
+#define _MBED_CONCAT_(a, b) a##b
```

Comment:
What file should these belong to do you think? toolchain.h? new file for preprocessor declarations? mbed_preprocessor.h?